### DISCUSSION TITLE
How to get the remote publickey(node_id)?

### DISCUSSION THREAD

**Discussion by lcp2000k** [n0-computer/iroh#3078] [Q&A] [answered]
when the endpoint accept an incomming,How to get the remote publickey(node_id)?

**Comment by dignifiedquire** [comment, marked as answer]
This is only available, once cthe `Incoming` has become a `Connection`, at which point you can use [`get_remote_node_id`](https://docs.rs/iroh/latest/iroh/endpoint/fn.get_remote_node_id.html)

You can see an example of this here: https://github.com/n0-computer/iroh/blob/main/iroh/examples/echo.rs#L80
